src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/TransformFunctions.kt

```diff
@@ -75,3 +81,22 @@ fun transform(sentenceAdjustments: SentenceAdjustments): MutableMap<AdjustmentTy
   adjustments[UNLAWFULLY_AT_LARGE] = sentenceAdjustments.unlawfullyAtLarge
   return adjustments
 }
+
+fun transform(booking: Booking, username: String, calculationStatus: CalculationStatus): CalculationRequest {
+  return CalculationRequest(
+    prisonerId = booking.offender.reference,
+    bookingId = booking.bookingId,
+    calculationStatus = calculationStatus.name,
+    calculatedByUsername = username,
+//    TODO mapping to JSON needs a spike as this doesnt work out of the box
+//    inputData = JacksonUtil.toJsonNode(mapper.writeValueAsString(booking))
+  )
```

Comment:
TODO here - will spike the conversion to JSON. At the moment having issues with `lateinit vars`

Reply:
This has been fixed following @timharrison-moj suggecstion to add `@JsonIgnore` - thanks @timharrison-moj  !